tools/crm_error.c

```diff
@@ -102,17 +102,15 @@ main(int argc, char **argv)
     }
 
     if (options.do_list) {
-        int start, end, width;
+        int start, end;
 
         // 256 is a hacky magic number that "should" be enough
         if (options.as_rc) {
             start = pcmk_rc_error - 256;
             end = PCMK_CUSTOM_OFFSET;
```

Comment:
If we're going to try to improve this, we might as well go further and choose start more reliably. We could define a new alias enum value for the lowest-numbered value (with a loud comment reminding us to update it), or you could loop until you hit the default message.

end is probably impossible to determine reliably since it's the largest system errno, and there's no symbol defined for that and no guarantee they're consecutive. I'm not sure if there's a better approach than this.

Reply:
I did that in the (currently) second-to-last commit in this PR. Title in case the commit itself gets overwritten: "Refactor: libpacemaker: Improve loop bounds in pcmk__list_result_codes()"

Reply:
I just moved all the refactor and non-API feature stuff to the beginning of the commit chain and did them in `crm_error.c` instead of in `pcmk_result_code.c` and `pcmk_output.c`, for easier review.

Reply:
Let's split this into two PRs at "Use formatted output", this one has gotten difficult to follow

Reply:
I misread you there and moved the stuff *after* formatted output. I'll move the formatted output commits over there shortly.